lib/v2/screens/profile_screens/profile/components/citizenship_card.dart

```diff
@@ -127,7 +127,16 @@ class CitizenshipCard extends StatelessWidget {
                                       ),
                                     ),
                                     const Expanded(child: SizedBox(width: 6)),
-                                    CitizenshipUpgradeButton(state.citizenshipUpgradeStatus),
+                                    CitizenshipUpgradeButton(
+                                      citizenshipUpgradeStatus: state.citizenshipUpgradeStatus,
+                                      onPressed: () {
+                                        state.citizenshipUpgradeStatus == CitizenshipUpgradeStatus.canResident
+                                            ? BlocProvider.of<ProfileBloc>(context)
+                                                .add(const OnActivateResidentButtonTapped())
+                                            : BlocProvider.of<ProfileBloc>(context)
+                                                .add(const OnActivateCitizenButtonTapped());
```

Comment:
I would have made it a single event (`OnUpgradeUserButtonTapped`) to avoid so much vervosity here and then in the bloc check the upgrade status, but also with this form you can see the most detailed event in the log.